### DISCUSSION TITLE
Transition state

### DISCUSSION THREAD

**Discussion by microomran** [SSCHAcode/python-sscha#58] [General]
Dear Lorenzo,

First thank you for this fine piece of work. Did you try to apply SSCHA to calculate gibbs(helmhotlz) free energy of T.S.?
In your first example you showed that a harmonic hamiltonian must be positive definite to generate a density matrix. As you probably aware in case of T.S. , there is always one imaginary frequency. 

Thank you
Regards,
Ahmed OMRAN

**Comment by mesonepigreco** [comment]
Dear Ahmed OMRAN,
I'm delighted you appreciated the SSCHA code. 
The code can be used to study highly anharmonic systems, even those that have harmonic imaginary frequencies. For example, this is the case in the tutorial for SnTe:

[www.sscha.eu/Tutorials/SnTe/](http://sscha.eu/Tutorials/SnTe)

We also provide the force field for this system, so you can reproduce all the calculations on your laptop without the need to run heavy ab-initio simulations.

In this case, the harmonic dynamical matrix is not positive definite. The SCHA needs a positive dynamical matrix to describe the density matrix of the system, enforced by:
```
dyn_sscha.ForcePositiveDefinite()
```
This auxiliary dynamical matrix describes the mean average anharmonic thermal/quantum fluctuations around the centroid positions.
If you want the anharmonic phonon frequencies (which may be imaginary if the system is unstable), you can get them as the free energy hessian. The tutorial on SnTe analyzes this aspect in the last part (The instability).

I have not studied the system you mention, but I can assume the guide on SnTe could be a similar system from which you can start to do a calculation.

I hope this was helpful,
Bests,
Lorenzo

**Comment by microomran** [comment]
Thank you Lorenzo. I will go through that example.

Regards/Cordialement,
Omran
